### PR TITLE
[#5181] feat(oceanbase-catalog): Add implementation for converter package

### PR DESCRIPTION
**Pull request by yuanoOo**
### What changes were proposed in this pull request?

- Add implementation for converter package:
  - `OceanBaseColumnDefaultValueConverter`
  - `OceanBaseExceptionConverter`
  - `OceanBaseTypeConverter`

### Why are the changes needed?

Fix: #5181 

### Does this PR introduce _any_ user-facing change?

no

### How was this patch tested?

Add unit test: `TestOceanBaseTypeConverter`

### CODE REVIEW

File: catalogs/catalog-jdbc-oceanbase/src/main/java/org/apache/gravitino/catalog/oceanbase/converter/OceanBaseColumnDefaultValueConverter.java
Comment (mchades) [inline catalogs/catalog-jdbc-oceanbase/src/main/java/org/apache/gravitino/catalog/oceanbase/converter/OceanBaseColumnDefaultValueConverter.java:1]: Is there any difference between these converters and MySQL? If not, I suggest making it the default implementation of the JDBC converter as well.

Comment (yuanoOo) [reply catalogs/catalog-jdbc-oceanbase/src/main/java/org/apache/gravitino/catalog/oceanbase/converter/OceanBaseColumnDefaultValueConverter.java:1]: There are some differences with MySQL, mainly in these data types:
```
      case OceanBaseTypeConverter.NUMBER:
      case OceanBaseTypeConverter.NUMERIC:
      case OceanBaseTypeConverter.DECIMAL:
        return Literals.decimalLiteral(
            Decimal.of(
                columnDefaultValue,
                Integer.parseInt(type.getColumnSize()),
                Integer.parseInt(type.getScale())));
      case OceanBaseTypeConverter.JSON:
      case JdbcTypeConverter.TEXT:
        return Literals.stringLiteral(columnDefaultValue);
```

**Comment by yuanoOo** [comment]
> Please update the PR title to comply with the format.

Fixed.

File: catalogs/catalog-jdbc-oceanbase/src/main/java/org/apache/gravitino/catalog/oceanbase/converter/OceanBaseColumnDefaultValueConverter.java
Comment (mchades) [reply catalogs/catalog-jdbc-oceanbase/src/main/java/org/apache/gravitino/catalog/oceanbase/converter/OceanBaseColumnDefaultValueConverter.java:1]: what about TypeConverter and ExceptionConverter?

File: catalogs/catalog-jdbc-oceanbase/src/main/java/org/apache/gravitino/catalog/oceanbase/converter/OceanBaseTypeConverter.java [hunk 1]
```diff
@@ -19,22 +19,132 @@
...
+      case NUMBER:
+      case NUMERIC:
+      case DECIMAL:
+        return Types.DecimalType.of(
+            Integer.parseInt(typeBean.getColumnSize()), Integer.parseInt(typeBean.getScale()));
+      case VARCHAR:
+        return Types.VarCharType.of(Integer.parseInt(typeBean.getColumnSize()));
+      case CHAR:
+        return Types.FixedCharType.of(Integer.parseInt(typeBean.getColumnSize()));
+      case JSON:
```
Comment (mchades) [inline catalogs/catalog-jdbc-oceanbase/src/main/java/org/apache/gravitino/catalog/oceanbase/converter/OceanBaseTypeConverter.java:86]: suggest removing this, we can add this after Gravitino supports `JSON` data type

File: catalogs/catalog-jdbc-oceanbase/src/main/java/org/apache/gravitino/catalog/oceanbase/converter/OceanBaseColumnDefaultValueConverter.java
Comment (yuanoOo) [reply catalogs/catalog-jdbc-oceanbase/src/main/java/org/apache/gravitino/catalog/oceanbase/converter/OceanBaseColumnDefaultValueConverter.java:1]: TypeConverter is similar to ColumnDefaultValueConverter above.
ExceptionConverter will be refactored to the default implementation.

File: catalogs/catalog-jdbc-oceanbase/src/main/java/org/apache/gravitino/catalog/oceanbase/converter/OceanBaseTypeConverter.java (hunk 1 above)
Comment (yuanoOo) [reply catalogs/catalog-jdbc-oceanbase/src/main/java/org/apache/gravitino/catalog/oceanbase/converter/OceanBaseTypeConverter.java:86]: Removed.

File: catalogs/catalog-jdbc-oceanbase/src/test/java/org/apache/gravitino/catalog/oceanbase/converter/TestOceanBaseTypeConverter.java [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+import org.apache.gravitino.rel.types.Types;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Test class for {@link org.apache.gravitino.catalog.oceanbase.converter.OceanBaseTypeConverter}
+ */
+public class TestOceanBaseTypeConverter {
+
+  private static final OceanBaseTypeConverter MYSQL_TYPE_CONVERTER = new OceanBaseTypeConverter();
```
Comment (mchades) [inline catalogs/catalog-jdbc-oceanbase/src/test/java/org/apache/gravitino/catalog/oceanbase/converter/TestOceanBaseTypeConverter.java:47]: `MYSQL_TYPE_CONVERTER` -> `OCEANBASE_TYPE_CONVERTER`

Comment (yuanoOo) [reply catalogs/catalog-jdbc-oceanbase/src/test/java/org/apache/gravitino/catalog/oceanbase/converter/TestOceanBaseTypeConverter.java]: Fixed.

**Review by mchades — APPROVED** [review]
LGTM. Thanks for your contributions!

**Pull request merged**